app/src/processing/app/contrib/ContributionManagerDialog.java

```diff
@@ -485,129 +511,149 @@ public void setProgress(int value) {
 //        int percent = 100 * value / this.max;
         progressBar.setValue(value);
         progressBar.setStringPainted(true);
-        status.setMessage(Language.text("contrib.status.downloading_list"));
+        activeTab.status.setMessage(Language
+          .text("contrib.status.downloading_list"));
       }
 
       @Override
       public void finishedAction() {
         progressBar.setVisible(false);
+        activeTab.updateContributionListing();
+        activeTab.updateCategoryChooser();
 
-        updateContributionListing();
-        updateCategoryChooser();
-
-        retryConnectingButton.setEnabled(true);
+        activeTab.retryConnectingButton.setEnabled(true);
 
         if (error) {
           if (exception instanceof SocketTimeoutException) {
-            status.setErrorMessage(Language
+            activeTab.status.setErrorMessage(Language
               .text("contrib.errors.list_download.timeout"));
           } else {
-            status.setErrorMessage(Language
+            activeTab.status.setErrorMessage(Language
               .text("contrib.errors.list_download"));
           }
           exception.printStackTrace();
-          retryConnectingButton.setVisible(true);
+          activeTab.retryConnectingButton.setVisible(true);
 
         } else {
-          status.setMessage(Language.text("contrib.status.done"));
-          retryConnectingButton.setVisible(false);
+          activeTab.status.setMessage(Language.text("contrib.status.done"));
+          activeTab.retryConnectingButton.setVisible(false);
         }
       }
     });
   }
 
-
-  protected void setFilterText(String filter) {
-    if (filter == null || filter.isEmpty()) {
-      filterField.setText("");
-      filterField.showingHint = true;
-    } else {
-      filterField.setText(filter);
-      filterField.showingHint = false;
+  /**
+   * 
+   * @return the currently selected tab
+   */
+  private ContributionTab getActiveTab() {
+    
+    switch (tabbedPane.getSelectedIndex()) {
+    case 0:
+      return toolsContributionTab;
+    case 1:
+      return librariesContributionTab;
+    case 2:
+      return modesContributionTab;
+    case 3:
+      return examplesContributionTab;
+    default:
+      return updatesContributionTab;
     }
-    filterField.applyFilter();
+    
   }
 
-
-//  private JPanel getPlaceholder() {
-//    return contributionListPanel.statusPlaceholder;
+//
```

Comment:
Just use /* block comments */ for large sections like this (that don't already have block comments), so it doesn't mark every single line as changed/rewritten.